libcloud/compute/drivers/dimensiondata.py

```diff
@@ -405,6 +442,163 @@ def ex_list_network_domains(self, location=None):
                                       params=params).object
         return self._to_network_domains(response)
 
+    def ex_create_network_domain(self, location, name, service_plan,
+                                 description=None):
+        """
+        Deploy a new network domain to a data center
+        """
+        create_node = ET.Element('deployNetworkDomain', {'xmlns': TYPES_URN})
+        ET.SubElement(create_node, "datacenterId").text = location.id
+        ET.SubElement(create_node, "name").text = name
+        if description is not None:
+            ET.SubElement(create_node, "description").text = description
+        ET.SubElement(create_node, "type").text = service_plan
+
+        response = self.connection.request_with_orgId_api_2(
+            'network/deployNetworkDomain',
+            method='POST',
+            data=ET.tostring(create_node)).object
+
+        network_domain_id = None
+
+        for info in findall(response, 'info', TYPES_URN):
```

Comment:
I believe you could also use a more complex xpath expression to directly match the `networkDomainID` attribute to avoid for loop, but I'm also fine with leaving it as is.